agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4730,6 +4734,26 @@ bool slave_thread::handle_ap_metrics_query(Socket *sd, ieee1905_1::CmduMessageRx
 
 }
 
+bool slave_thread::handle_link_metrics_query(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid = cmdu_rx.getMessageId();
+
+    auto tlv_type = cmdu_rx.getNextTlvType();
+    while (tlv_type != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
```

Comment:
IEEE1905.1 says that there should be only one link metric query TLV. So while loop is not necessary here either. 
Use addClass and return false if it fails.

Reply:
done

Reply:
After some digging, I see that I was wrong. Apparently, every CMDU could have VS_TLV inside it, and it can be first.
It means that you must use the while loop as you did, and add a handler for VS_TLV.